system-parachains/coretime/coretime-polkadot/src/migrations.rs

```diff
@@ -0,0 +1,330 @@
+			.into_iter()
+			.enumerate()
+		{
+			assert_eq!(
+				Workplan::<Runtime>::get((workplan_start, core_id as u16)),
+				Some(Schedule::truncate_from(Vec::from([ScheduleItem {
+					mask: CoreMask::complete(),
+					assignment: task,
+				}])))
+			);
+		}
+
+		// Make sure we've got all the leases.
+		let leases = Leases::<Runtime>::get();
+		assert_eq!(leases.len(), LEASES.iter().filter(|(_, l)| sale_info.region_end <= *l).count());
+
+		// Make something out of the leases that is easier to check against.
+		let leases_vec: Vec<(u32, u32)> =
+			leases.iter().map(|LeaseRecordItem { until, task }| (*task, *until)).collect();
+
+		// Iterate through hardcoded leases and check they're all correctly in state and scheduled
+		// in the workplan.
+		for (i, (para_id, until)) in LEASES.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before the leases.
+			let core_id = i as u16 + 4;
+
+			assert!(leases_vec.contains(&(*until, *para_id)));
+
+			// This is the entry found in Workplan
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// They should all be in the workplan for next region.
+			assert_eq!(Workplan::<Runtime>::get((workplan_start, core_id)), Some(workload));
+		}
+
+		// For the leases we had before their lease should extend for an additional 11520
+		// timeslices (64 days).
+		for LeaseRecordItem { task, until } in prev_leases.iter() {
+			log::error!("{task}, {until}");
+			assert!(leases_vec.contains(&(*until + 11520, *task)))
+		}
+
+		// Iterate through hardcoded potential renewals and check they're all correctly in state
+		// and scheduled in the workplan.
+		for (i, (para_id, until)) in POTENTIAL_RENEWALS.iter().enumerate() {
+			// Add the system parachains as an offset - these should come before
+			// the leases.
+			let core_id = i as u16 + 4;
+
+			// This is the entry found in Workplan and PotentialRenewals.
+			let workload = Schedule::truncate_from(Vec::from([ScheduleItem {
+				mask: CoreMask::complete(),
+				assignment: Task(*para_id),
+			}]));
+
+			// Make sure they're not in the leases.
+			assert!(!leases.contains(&LeaseRecordItem { until: *until, task: *para_id }));
+
+			// Ensure they can renew in the next region.
+			assert_eq!(
+				PotentialRenewals::<Runtime>::get(PotentialRenewalId {
+					core: core_id,
+					when: sale_info.region_end + 5040,
+				}),
+				Some(PotentialRenewalRecord {
+					price: 1_000_000_000_000,
+					completion: pallet_broker::CompletionStatus::Complete(workload.clone())
+				})
+			);
+
+			// They should all be in the workplan for next sale.
+			assert_eq!(Workplan::<Runtime>::get((workplan_start, core_id)), Some(workload));
+		}
+
+		// Walk the workplan at timeslice 287565 and make sure there is an entry for every 62 cores.
+		for i in 0..61 {
+			// TODO - finalise
+			assert!(Workplan::<Runtime>::get((287565, i)).is_some());
```

Comment:
There is no entry for core 55 which should prove that the range
https://github.com/polkadot-fellows/runtimes/pull/458/files#diff-f14664ec4b1b7de51605453415a9d92013894e44aa7ee3f9f65277c6c71c8a57R65
is wrong and should be:
```rust
for (&(para_id, _), core_id) in LEASES.iter().zip(51u16..56u16)
```

Reply:
Those are the cores on the open market, which can either have an entry or not depending on whether somebody has assigned it yet